### PR TITLE
The Big Renaming

### PR DESCRIPTION
**Pull request by fingolfin**
Adjust to renamings in AbstractAlgebra, Nemo, Hecke

WIP: the tests probably will fail, and I need to look into a few other things.

### CODE REVIEW

**Comment by thofma** [comment]
I added a commit with 3 lines of changes that @StevellM told me about.

**Comment by fingolfin** [comment]
@thofma @StevellM hmmm
```
Orthogonal groups of non-semiregular torquadmod: Error During Test at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/test/Groups/abelian_aut.jl:112
  Test threw exception
  Expression: order(orthogonal_group(TT)) == 24
  UndefVarError: `biproduct` not defined
  Stacktrace:
    [1] _compute_gens_non_split_degenerate(T::TorQuadModule)
```

**Comment by fingolfin** [comment]
I've reverted that patch for now, and added another fix of mine. Unfortunately my laptop crashed earlier so all is delayed *sigh*

**Comment by thofma** [comment]
Ah oops, my bad. That PR was not merged yet, I was confused.

**Comment by fingolfin** [comment]
@fieker @thofma any idea about this one:
```
Experimental.galois_group: Error During Test at /Users/mhorn/Projekte/OSCAR/Oscar.spielwiese/test/Experimental/galois-test.jl:1
  Got exception outside of a @test
  MethodError: no method matching ComplexField(::Int64)

  Closest candidates are:
    ComplexField()
     @ Nemo ~/.julia/packages/Nemo/gOKbp/src/arb/ArbTypes.jl:227

  Stacktrace:
    [1] galois_group(K::AnticNumberField, extra::Int64; useSubfields::Bool, pStart::Int64, prime::Int64, algo::Symbol)
      @ Oscar.GaloisGrp ~/Projekte/OSCAR/Oscar.spielwiese/experimental/GaloisGrp/GaloisGrp.jl:1938
    [2] kwcall(::NamedTuple{(:algo,), Tuple{Symbol}}, ::typeof(galois_group), K::AnticNumberField)
      @ Oscar.GaloisGrp ~/Projekte/OSCAR/Oscar.spielwiese/experimental/GaloisGrp/GaloisGrp.jl:1911
    [3] macro expansion
      @ ~/Projekte/OSCAR/Oscar.spielwiese/test/Experimental/galois-test.jl:29 [inlined]
    [4] macro expansion
      @ ~/.julia/juliaup/julia-1.9.0-beta4+0.aarch64.apple.darwin14/share/julia/stdlib/v1.9/Test/src/Test.jl:1496 [inlined]
    [5] top-level scope
      @ ~/Projekte/OSCAR/Oscar.spielwiese/test/Experimental/galois-test.jl:3
    [6] include
      @ ./Base.jl:458 [inlined]
    [7] macro expansion
      @ ./timing.jl:501 [inlined]
    [8] include(str::String)
      @ Main ~/Projekte/OSCAR/Oscar.spielwiese/test/runtests.jl:56
    [9] top-level scope
      @ ~/Projekte/OSCAR/Oscar.spielwiese/test/runtests.jl:102
   [10] include(fname::String)
      @ Base.MainInclude ./client.jl:478
   [11] top-level scope
      @ none:6
   [12] eval
      @ ./boot.jl:370 [inlined]
   [13] exec_options(opts::Base.JLOptions)
      @ Base ./client.jl:280
   [14] _start()
      @ Base ./client.jl:522
```

**Comment by fingolfin** [comment]
.... and now I need to redo it to accommodate for the merge of @HereAround's PR (which is fine, I'll do it later)

**Comment by HereAround** [comment]
Sorry @fingolfin . I hope it does not cause too much trouble?

**Comment by thofma** [comment]
I can do the change `ComplexField` in a separate PR if that is easier for you. Otherwise, `ComplexField` must be `AcbField`.

**Review by thofma — APPROVED** [review]
I fixed some other minor thing.

**Comment by afkafkafk13** [comment]
naive question: Is there going to be a CheatSheat that we can give e.g. to our students, who have already written fragments of code (e.g. for their Master Thesis), but need to rebase now. That would also be very helpful for every developer who was not directly involved in the renaming.

**Comment by thofma** [comment]
I would mention that at #1980

File: docs/oscar_references.bib [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ @Article{Shi15
   title         = {An algorithm to compute automorphism groups of {$K3$}
                   surfaces and an application to singular {$K3$} surfaces},
   mrnumber      = {3456710},
-  journal       = {Int. Math. Res. Not. IMRN},
+  journal       = {Int. Math. ResidueRingElem. Not. IMRN},
```
Comment (joschmitt) [inline docs/oscar_references.bib:1259]: That change wasn't intended, was it?

File: docs/src/General/faq.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ You can find our installation instructions [here](https://oscar.computeralgebra.
 
 ---
 
-**Q: Why do some of your types have funny names like `fmpz` or `fmpq_mat`?**
+**Q: Why do some of your types have funny names like `ZZRingElem` or `QQMatrix`?**
 
 This has historical reasons. We plan to rename these types before OSCAR 1.0
 (the old names will still work indefinitely, though).
```
Comment (lgoettgens) [inline docs/src/General/faq.md:19]: This faq entry is kind of outdated

**Comment by fingolfin** [comment]
Agreed to both. Pull requests would be most welcome, or at least issues recording these :-)

**Comment by lgoettgens** [comment]
I'll do a pr :)

File: docs/oscar_references.bib (hunk 1 above)
Comment (lgoettgens) [reply docs/oscar_references.bib:1259]: Resolved in #1998

File: docs/src/General/faq.md (hunk 2 above)
Comment (lgoettgens) [reply docs/src/General/faq.md:19]: Resolved in #1998

**Pull request merged**